libexec/functions

```diff
@@ -441,6 +441,66 @@ is_tar() {
 }
 
 
+bind_nvidia_files() {
+# either use nvidia-container-cli or nvliblist.conf to find nvidia libs and
+# binaries for binding at runtime with --nv 
+    if ! SINGULARITY_NVLIBLIST=`mktemp "${TMPDIR:-/tmp}"/.singularity-nvliblist.XXXXXXXX`; then
+        message ERROR "Failed to create temporary nvliblist file.\n"
+        ABORT 255
+    fi
+
+    # try to use nvidia-container-cli if installed and functional
+    if singularity_which nvidia-container-cli > /dev/null 2>&1 && SINGULARITY_NV_FILES=`nvidia-container-cli list -cguv`; then
+        # bins should have full paths
+        SINGULARITY_NVBINLIST=`echo "$SINGULARITY_NV_FILES" | grep -Ev "\.so"`
+        # libs should just be basenames ending in ".so"
+        for i in `echo "$SINGULARITY_NV_FILES" | grep -E "\.so"`; do
+            basename "$i" | awk -F "\.so" '{print $1 ".so"}' >> "$SINGULARITY_NVLIBLIST"
+        done
+
+    else # fall back to nvliblist.conf
+        message 2 "nvidia-container-cli not installed or returning error\n"
+        message 2 "falling back to $SINGULARITY_sysconfdir/singularity/nvliblist.conf\n"
+        # strip comments and spaces and bins do NOT end in .so
+        SINGULARITY_NVBINLIST=`cat $SINGULARITY_sysconfdir"/singularity/nvliblist.conf" | grep -Ev "^#|^\s*$" | grep -vE "\.so$"`
+        # strip comments and spaces and libs DO end in .so
+        cat "$SINGULARITY_sysconfdir/singularity/nvliblist.conf" | grep -Ev "^#|^\s*$" | grep -E "\.so$" > $SINGULARITY_NVLIBLIST
+    fi
```

Comment:
OCD dinosaur kicking in... can we do:

```
cat "${SINGULARITY_sysconfdir}/ ...etc
```
instead?

Reply:
Yes we can!

Reply:
https://www.youtube.com/watch?v=RF5ubSoXZGM